bot/user_service.py

```diff
@@ -0,0 +1,204 @@
+        # REMOVE STALE GROUPS
+        for ggroup, gname in current_guild_groups:
+            if ggroup in hidden_groups:
+                LOG.info("Player %s chose to hide group '%s', which is now removed.", auth.name, ggroup)
+                ts_facade.servergroup_client_del(servergroup_id=ts_groups[ggroup], client_db_id=client_db_id)
+            elif gname not in ingame_member_of:
+                if ggroup not in ts_groups:
+                    LOG.warning(
+                        "Player %s should be removed from the TS group '%s' because they are not a member of guild '%s'."
+                        " But no matching group exists."
+                        " You should remove the entry for this guild from the db or check the spelling of the TS group in the DB. Skipping.",
+                        ggroup, auth.name, gname)
+                else:
+                    LOG.info("Player %s is no longer part of the guild '%s'. Removing attached group '%s'.", auth.name, gname, ggroup)
+                    ts_facade.servergroup_client_del(servergroup_id=ts_groups[ggroup], client_db_id=client_db_id)
+
+        # ADD DUE GROUPS
+        for g in ingame_member_of:
+            ts_group = None
+            with self._database_connection.lock:
+                ts_group = self._database_connection.cursor.execute("SELECT ts_group FROM guilds WHERE guild_name = ?", (g,)).fetchone()
+            if ts_group:
+                ts_group = ts_group[0]  # first and only column, if a row exists
+                if ts_group not in current_group_names:
+                    if ts_group in hidden_groups:
+                        LOG.info("Player %s is entitled to TS group '%s', but chose to hide it. Skipping.", auth.name, ts_group)
+                    else:
+                        if ts_group not in ts_groups:
+                            LOG.warning(
+                                "Player %s should be assigned the TS group '%s' because they are member of guild '%s'."
+                                " But the group does not exist. You should remove the entry for this guild from the db or create the group."
+                                " Skipping.",
+                                auth.name, ts_group, g)
+                        else:
+                            LOG.info("Player %s is member of guild '%s' and will be assigned the TS group '%s'.", auth.name, g, ts_group)
+                            ts_facade.servergroup_client_add(servergroup_id=ts_groups[ts_group], client_db_id=client_db_id)
+
+    # Helps find the group ID for a group name
+    def _find_group_by_name(self, group_to_find):
+        with self._ts_connection_pool.item() as ts_facade:
+            self.groups_list = ts_facade.servergroup_list()
+        for group in self.groups_list:
+            if group.get('name') == group_to_find:
+                return group.get('sgid')
+        return -1
+
+    def clientNeedsVerify(self, unique_client_id):
```

Comment:
Camelcase (also all following methods)

Reply:
Those were just moved from the "Bot" Class.

Reply:
Fixed

Reply:
Fixed.